android/engine/src/main/java/org/smartregister/fhircore/engine/datastore/serializers/PractitionerDetailsDataStoreSerializer.kt

```diff
@@ -0,0 +1,34 @@
+package org.smartregister.fhircore.engine.datastore.serializers
+
+import androidx.datastore.core.Serializer
+import kotlinx.serialization.json.Json
+import org.apache.commons.lang3.SerializationException
+import org.smartregister.fhircore.engine.datastore.mockdata.SerializablePractitionerDetails
+import java.io.InputStream
+import java.io.OutputStream
+
+object PractitionerDetailsDataStoreSerializer: Serializer<SerializablePractitionerDetails> {
+    override val defaultValue: SerializablePractitionerDetails
+        get() = SerializablePractitionerDetails()
+
+    override suspend fun readFrom(input: InputStream): SerializablePractitionerDetails {
+        return try {
+            Json.decodeFromString(
+                deserializer = SerializablePractitionerDetails.serializer(),
+                string = input.readBytes().decodeToString()
+            )
+        } catch (e: SerializationException) {
+            e.printStackTrace()
```

Comment:
Log using timber

Reply:
Can when get other exceptions while decoding the JSON?  Please catch those too

Reply:
> Can when get other exceptions while decoding the JSON? Please catch those too

No. Only serialization exception
![image](https://github.com/opensrp/fhircore/assets/129178453/5cd6b1bc-e386-4a61-ab27-3503ec530393)

Reply:
> Log using timber

done